### PR TITLE
Mahdhi/new lib for mysql

### PR DESCRIPTION
**Pull request by mabroukmahdhi**
# Introduction
I decided to implement those changes because I liked the library and wanted to use it within a MySQL connection. That was not possible without new implementations.
If you try to connect to MySql server you will get this exception #31 .

# Changes
I tried to write as long as posible a shared code. 

**Example**

Now the `EFxceptionsContext `for `SqlServer ` looks like :
```c#
public class EFxceptionsContext : BaseContext<SqlException>
    {
        protected override IDbErrorBroker<SqlException> CreateErrorBroker()
        {
            return new SqlErrorBroker();
        }

        protected override IEFxceptionService CreateEFxceptionService(IDbErrorBroker<SqlException> errorBroker)
        {
            return new EFxceptionService<SqlException>(errorBroker);
        }

    }
```
And for `MySql` it looks like:

```c#
 public class EFxceptionsContext : BaseContext<MySqlException>
    {
        protected override IDbErrorBroker<MySqlException> CreateErrorBroker()
        {
            return new MySqlErrorBroker();
        }

        protected override IEFxceptionService CreateEFxceptionService(IDbErrorBroker<MySqlException> errorBroker)
        {
            return new EFxceptionService<MySqlException>(errorBroker);
        }

    }
```
The `BaseContext` resp. `BaseIdentityContext ` is implemented in a separated class library named `EFxceptions.Core` resp. `EFxceptions.Identity.Core`.

Made w/ ❤️ 

And I hope that people find something interesting here.
Mabrouk Mahdhi

### CODE REVIEW

File: EFxceptions.Core/EFxceptions.Core.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
```
Comment (hassanhabib) [inline EFxceptions.Core/EFxceptions.Core.csproj:5]: disable this and the one below it. build again and fix the errors

File: EFxceptions.Identity.MySql.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using Xunit;
+
+namespace EFxceptions.Identity.MySql.Tests
+{
+    public class UnitTest1
+    {
+        [Fact]
+        public void Test1()
```
Comment (hassanhabib) [inline EFxceptions.Identity.MySql.Tests/UnitTest1.cs:8]: Remove this file

**Comment by mabroukmahdhi** [comment]
@hassanhabib hmm as I told you I am going to create a new PR soon. It will be better than this ugly one.

**Comment by mabroukmahdhi** [comment]
Duplicate. Please check #34

**Pull request closed without merging**